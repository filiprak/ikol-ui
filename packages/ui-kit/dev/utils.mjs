import { existsSync } from 'fs';
import { readFile, writeFile, mkdir } from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

export async function useTemplateFile(filename, outputFilepath, variables) {
  const data = await readFile(path.resolve(__dirname, 'templates', filename), 'utf8');
  return useTemplate(data, outputFilepath, variables);
}

export async function useTemplate(templateStr, outputFilepath, variables) {
  const updatedTemplate = templateStr.replace(/\[\[([a-zA-Z0-9_]+)\]\]/g, (match, key) => {
    return variables[key] || match;
  });

  if (!existsSync(path.dirname(outputFilepath))) {
    await mkdir(path.dirname(outputFilepath));
  }
  await writeFile(outputFilepath, updatedTemplate);
}

export const kebabize = str => {
  return str.split('').map((letter, idx) => {
      return letter.toUpperCase() === letter
          ? `${idx !== 0 ? '-' : ''}${letter.toLowerCase()}`
          : letter;
  }).join('');
}

export const ROOT_DIR = path.resolve(__dirname, '..');
