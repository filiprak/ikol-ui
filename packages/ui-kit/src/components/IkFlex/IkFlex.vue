<script setup lang="ts">
import '@ui/styles';
import './IkFlex.css';
import { computed, h, useSlots } from 'vue';
import { useRender } from '@ui/composables/render';

const props = defineProps({
    tag: {
        type: String,
        default: 'div',
    },
    column: {
        type: Boolean,
        default: false,
    },
    inline: {
        type: Boolean,
        default: false,
    },
    full_height_items: {
        type: Boolean,
        default: false,
    },
    full_width_items: {
        type: Boolean,
        default: false,
    },
    wrap: {
        type: Boolean,
        default: false,
    },
    wrap_text: {
        type: Boolean,
        default: false,
    },
    justify_start: {
        type: Boolean,
        default: false,
    },
    justify_end: {
        type: Boolean,
        default: false,
    },
    justify_space: {
        type: Boolean,
        default: false,
    },
    align_start: {
        type: Boolean,
        default: false,
    },
    align_end: {
        type: Boolean,
        default: false,
    },
    fixed_and_grow: {
        type: Boolean,
        default: false,
    },
    grow_and_fixed: {
        type: Boolean,
        default: false,
    },
    reverse: {
        type: Boolean,
        default: false,
    },
});
const slots = useSlots();
const classes = computed(() => {
    const classes = ['ik-flex'];

    if (props.column) {
        classes.push('ik-flex--column');
    }
    if (props.reverse) {
        classes.push('ik-flex--reverse');
    }
    if (props.inline) {
        classes.push('ik-flex--inline');
    }
    if (props.full_width_items) {
        classes.push('ik-flex--full-width-items');
    }
    if (props.full_height_items) {
        classes.push('ik-flex--full-height-items');
    }
    if (props.wrap) {
        classes.push('ik-flex--wrap');
    }
    if (props.wrap_text) {
        classes.push('ik-flex--wrap-text');
    }
    if (props.justify_start) {
        classes.push('ik-flex--justify-start');
    }
    if (props.justify_space) {
        classes.push('ik-flex--justify-space');
    }
    if (props.justify_end) {
        classes.push('ik-flex--justify-end');
    }
    if (props.align_start) {
        classes.push('ik-flex--align-start');
    }
    if (props.align_end) {
        classes.push('ik-flex--align-end');
    }
    if (props.fixed_and_grow) {
        classes.push('ik-flex--fixed-grow');
    }
    if (props.grow_and_fixed) {
        classes.push('ik-flex--grow-fixed');
    }
    return classes;
});
useRender(() => h(props.tag, { class: classes.value }, slots.default?.()));
</script>
