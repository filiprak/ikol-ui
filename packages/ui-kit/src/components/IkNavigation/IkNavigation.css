.ik-navigation {
    position: fixed;
    left: 0;
    top: 0;
    z-index: var(--z-index-app-nav);
    display: flex;
    flex-direction: column;

    background-color: var(--secondary-color);
    color: var(--on-secondary-color);
    transition: width 0.2s ease-in-out, transform 0.2s ease-in-out;
}

.ik-theme--dark .ik-navigation:not(.ik-navigation--secondary) {
    background-color: var(--surface-color);
    color: var(--on-surface-color-1);
}

.ik-navigation--secondary {
    position: relative;
    z-index: initial;
    background: none;
    width: 250px;
    color: var(--on-background-color-3);
    transition: none;
    max-height: 100%;
}

.ik-navigation__content {
    display: flex;
    flex-direction: column;
    flex: 1 1 auto;
    min-height: 0;
}

.ik-navigation__backdrop {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: #000;
    opacity: .5;
    z-index: 3;
}

.ik-theme--dark .ik-navigation__backdrop {
    background-color: #4e4e4e;
}

.ik-navigation--secondary .ik-navigation__content {
    overflow: auto;
}

.ik-navigation--mobile.ik-navigation--secondary .ik-navigation__content {
    flex-direction: row;
    overflow: auto;
}

.ik-navigation__top,
.ik-navigation__bottom {
    flex: 0 0 auto;
}

.ik-navigation-item__link {
    position: relative;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-direction: row;
    padding: 0 20px 15px;
    outline: none;
    text-decoration: none !important;
}

.ik-navigation--mini .ik-navigation-item__link {
    text-align: center;
    padding: 0 10px 15px;
    justify-content: center;
}

.ik-navigation-item__link,
.ik-navigation-item__link:active,
.ik-navigation-item__link:focus,
.ik-navigation-item__link:visited,
.ik-navigation-item__link:hover {
    outline-width: 0;
    color: var(--on-surface-color-3);
}

.ik-navigation--secondary .ik-navigation-item__link {
    flex-direction: row;
    color: inherit;
    height: 48px;
    padding: 0 16px 0 15px;
    margin: 0;
}

.ik-navigation-item--clickable {
    cursor: pointer;
    user-select: none;
}

.ik-navigation-item__icon {
    flex: 0 0 36px;
    width: 36px;
    height: 36px;
    font-size: 15px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    transition: opacity 0.1s;
    -webkit-transition: opacity 0.1s;
}

.ik-navigation-item--circle-icon > .ik-navigation-item__link > .ik-navigation-item__icon {
    border-radius: 50%;
    background-color: rgba(255, 255, 255, 0.08);
}

.ik-navigation-item--active > .ik-navigation-item__link,
.ik-navigation-item--active > .ik-navigation-item__link:active,
.ik-navigation-item--active > .ik-navigation-item__link:hover,
.ik-navigation-item--active > .ik-navigation-item__link:focus,
.ik-navigation-item--active > .ik-navigation-item__link:visited {
    color: var(--on-primary-color);
}

.ik-navigation:not(.ik-navigation--secondary) .ik-navigation-item--active > .ik-navigation-item__link > .ik-navigation-item__icon {
    background-color: var(--primary-color);
}

.ik-navigation--secondary .ik-navigation-item--active > .ik-navigation-item__link,
.ik-navigation--secondary .ik-navigation-item--clickable > .ik-navigation-item__link:hover {
    background-color: var(--primary-color-o1);
    color: var(--primary-color);
}

.ik-navigation--mobile.ik-navigation--secondary .ik-navigation-item--active {
    border-bottom: 3px solid var(--primary-color);
}

.ik-navigation--mobile.ik-navigation--secondary .ik-navigation-item--active,
.ik-navigation--mobile.ik-navigation--secondary .ik-navigation-item--clickable:hover {
    background-color: var(--background-color) !important;
}

.ik-navigation--mini .ik-navigation-item__content {
    font-size: 10px;
    padding: 5px 0;
}

.ik-navigation-item__content {
    padding-left: 10px;
    flex-grow: 1;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
}

.ik-navigation--mobile.ik-navigation--secondary .ik-navigation-item__content {
    padding: 0;
}

.ik-navigation-item--clickable:not(.ik-navigation-item--active):hover .ik-navigation-item__icon {
    opacity: 0.7;
}

.ik-navigation-item__popover {
    background: var(--surface-color);
    color: var(--on-surface-color-1);
    box-shadow: 0 0 20px 8px rgba(31, 44, 56, 0.15);
    border-radius: 16px;
    overflow: hidden;
}

.ik-navigation:not(.ik-navigation--mini) .ik-navigation-item__children {
    padding: 10px 0;
    margin-bottom: 15px;
    background-color: rgba(0, 0, 0, 0.2);
}

.ik-navigation:not(.ik-navigation--mini) .ik-navigation-item__children .ik-divider__line {
    border-color: rgba(255, 255, 255, 0.1);
}

.ik-navigation-item__popover .ik-navigation-item__children .ik-divider {
    padding: 0 15px;
}

.ik-navigation-item__popover .ik-navigation-item__children .ik-divider__line {
    border-color: var(--border-color-2);
}

.ik-navigation-item__children .ik-navigation-item__link {
    display: flex;
    flex-direction: row;
    height: 42px;
    padding: 0 16px 0 66px;
    margin: 0;
    text-align: left;
}

.ik-navigation-item__popover .ik-navigation-item__link {
    padding: 0 16px 0 15px;
}

.ik-navigation-item__children .ik-navigation-item__top > .ik-navigation-item__link {
    color: var(--on-surface-color-1);
}

.ik-navigation-item__children .ik-navigation-item__content {
    font-size: 13px;
    padding: 0;
}

.ik-navigation-item__children .ik-navigation-item:hover > .ik-navigation-item__link,
.ik-navigation-item__children .ik-navigation-item--active > .ik-navigation-item__link {
    background-color: var(--primary-color-o1);
    color: var(--primary-color);
}

.ik-navigation-item__children .ik-navigation-item__icon {
    background: none !important;
    width: 28px;
    height: 28px;
    justify-content: flex-start;
}

.ik-navigation-item__arrow {
    min-width: 0;
}
