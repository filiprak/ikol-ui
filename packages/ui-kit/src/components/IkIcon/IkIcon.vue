<script setup lang="ts">
import '@ui/styles';
import './IkIcon.css';
import { type PropType, h, useAttrs } from 'vue';
import { toCssUnits, getIconClasses } from '@ui/utils/helpers';
import { useRender } from '@ui/composables/render';
import type { IkUIDesignColor } from '@ui/types/utils';

const props = defineProps({
    tag: {
        type: String,
        default: 'i',
    },
    icon: {
        type: String,
    },
    size: {
        type: [Number, String],
    },
    size_px: {
        type: [Number, String],
    },
    disabled: {
        type: Boolean,
        default: false,
    },
    circle: {
        type: Boolean,
        default: false,
    },
    design: {
        type: String as PropType<`${IkUIDesignColor}`>,
    },
    variant: {
        type: String,
    },
});
const attrs = useAttrs();

useRender(() => {
    return h(props.tag, {
        class: [
            {
                'ik-icon': true,
                ['ik-icon--' + props.design]: !!props.design,
                ['ik-icon--' + props.variant]: !!props.variant,
                'ik-icon--circle': props.circle,
                'ik-icon--clickable': !props.disabled && !!attrs.onClick,
                'ik-icon--disabled': props.disabled,
                ['fa-' + props.size + 'x']: !!props.size,
            },
            getIconClasses(props.icon || ''),
        ],
        style: {
            width: props.size_px && props.circle ? toCssUnits(props.size_px, 'px') : null,
            height: props.size_px && props.circle ? toCssUnits(props.size_px, 'px') : null,
            fontSize: props.size_px ? toCssUnits(props.circle ? (Number(props.size_px) / 2) : props.size_px, 'px') : null,
        },
    });
});
</script>
