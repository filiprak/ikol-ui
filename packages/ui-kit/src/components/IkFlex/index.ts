import IkFlex from './IkFlex.vue';
export interface IkFlex extends InstanceType<typeof IkFlex> { };
export { IkFlex };
