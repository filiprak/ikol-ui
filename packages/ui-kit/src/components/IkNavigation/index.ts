import IkNavigation from './IkNavigation.vue';
import IkNavigationItem from './IkNavigationItem.vue';

export interface IkNavigation extends InstanceType<typeof IkNavigation> { };
export interface IkNavigationItem extends InstanceType<typeof IkNavigationItem> { };

export { IkNavigation };
export { IkNavigationItem };
