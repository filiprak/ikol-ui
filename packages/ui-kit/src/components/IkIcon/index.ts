import IkIcon from './IkIcon.vue';
export interface IkIcon extends InstanceType<typeof IkIcon> { };
export { IkIcon };
