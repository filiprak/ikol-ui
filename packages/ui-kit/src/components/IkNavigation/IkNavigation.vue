<script lang="ts">
import '@ui/styles';
import './IkNavigation.css';
import type { PropType } from 'vue';
import { defineComponent, h, unref, inject } from 'vue';

import type { IkApp } from '@ui/components/IkApp';
import { toCssUnits } from '@ui/utils/helpers';
import { IkScrollArea } from '@ui/components/IkScrollArea';
import { useDevice } from '@ui/composables/device';

export default defineComponent({
    provide() {
        return { navigation: this };
    },
    props: {
        open: {
            type: Boolean,
            default: false,
        },
        mini: {
            type: Boolean,
            default: false,
        },
        tag: {
            type: String,
            default: 'nav',
        },
        width: {
            type: Number,
            default: 256,
        },
        mini_width: {
            type: Number,
            default: 72,
        },
        exclude_close: {
            type: Array as PropType<HTMLElement[]>,
            default() {
                return [];
            },
        },
        secondary: {
            type: Boolean,
            default: false,
        },
    },
    setup() {
        return { app: inject<IkApp>('app'), device: useDevice() };
    },
    data() {
        return {
            backdrop_el: null as HTMLElement | null,
        };
    },
    computed: {
        root_class() {
            return {
                'ik-navigation': true,
                'ik-navigation--mini': this.mini,
                'ik-navigation--secondary': this.secondary,
                'ik-navigation--mobile': unref(this.device.mobile),
            };
        },
        styles() {
            if (!this.secondary) {
                return {
                    height: '100vh',
                    top: 0,
                    maxHeight: '100%',
                    transform: 'translateX(' + (this.open ? '0%' : '-100%') + ')',
                    width: toCssUnits(this.computed_width),
                };
            } else {
                return {
                    width: unref(this.device.mobile) ? '100%' : null,
                };
            }
        },
        computed_width() {
            if (this.mini) {
                return this.mini_width;
            } else {
                return this.width;
            }
        },
    },
    watch: {
        open: {
            immediate: true,
            handler(value) {
                if (unref(this.device.mobile) && !this.secondary) {
                    if (value) {
                        window.addEventListener('mousedown', this._onWindowMouseDown, true);
                    } else {
                        window.removeEventListener('mousedown', this._onWindowMouseDown, true);
                    }
                    this._toggleBackdrop(value);
                }
            },
        },
        mini: {
            immediate: true,
            handler(mini) {
                if (!unref(this.device.mobile) && !this.secondary) {
                    if (!mini) {
                        window.addEventListener('mousedown', this._onWindowMouseDown, true);
                    } else {
                        window.removeEventListener('mousedown', this._onWindowMouseDown, true);
                    }
                }
            },
        },
        'device.mobile.value'(mobile) {
            this.$emit('update:open', !mobile);
            this.$emit('update:mini', !mobile);

            if (!mobile) {
                this._toggleBackdrop(false);
            }
        },
    },
    created() {
        if (!this.secondary) {
            this.app && this.app.registerNav(this);
        }
    },
    mounted() {
        if (!this.secondary) {
            this._initBackdrop();
        }
    },
    beforeUnmount() {
        if (!this.secondary) {
            this.app && this.app.unregisterNav();
            window.removeEventListener('mousedown', this._onWindowMouseDown, true);
        }
        this._destroyBackdrop();
    },
    methods: {
        toggleMini(mini?: boolean) {
            this.$emit('update:mini', mini !== undefined ? !!mini : !this.mini);
        },
        toggle(value?: boolean) {
            if (!this.secondary) {
                if (unref(this.device.mobile)) {
                    this.$emit('update:open', value !== undefined ? !!value : !this.open);
                } else {
                    this.$emit('update:mini', value !== undefined ? !value : !this.mini);
                }
            }
        },
        _onWindowMouseDown(event: MouseEvent) {
            let should_close = this.$el !== event.target && !this.$el.contains(event.target as HTMLElement);

            // check if element not excluded
            if (should_close) {
                for (let i = 0; i < this.exclude_close.length; i++) {
                    if (this.exclude_close[i].contains(event.target as HTMLElement)) {
                        should_close = false;
                        break;
                    }
                }
            }

            if (should_close) {
                this.toggle(false);
                event.preventDefault();
                event.stopPropagation();
            }
        },
        _initBackdrop() {
            const backdrop_el = document.createElement('div');

            backdrop_el.classList.add('ik-navigation__backdrop');
            backdrop_el.style.display = 'none';

            this.backdrop_el = backdrop_el;

            document.body.appendChild(backdrop_el);
        },
        _toggleBackdrop(visible?: boolean) {
            if (this.backdrop_el) {
                this.backdrop_el.style.display = visible ? '' : 'none';
            }
        },
        _destroyBackdrop() {
            if (this.backdrop_el) {
                this.backdrop_el.remove();
                this.backdrop_el = null;
            }
        },
    },
    render() {
        return h(this.tag, {
            class: this.root_class,
            style: this.styles,
        }, {
            default: () => {
                const content = h('div', { class: 'ik-navigation__content' }, [
                    this.secondary && this.device.mobile.value ? this.$slots.default?.() : h(IkScrollArea, {}, { default: () => this.$slots.default?.() }),
                ]);

                const top = h('div', { class: 'ik-navigation__top' }, [
                    this.$slots.top?.(),
                ]);
                const bottom = h('div', { class: 'ik-navigation__bottom' }, [
                    this.$slots.bottom?.(),
                ]);

                return [top, content, bottom];
            },
        });
    },
});
</script>
