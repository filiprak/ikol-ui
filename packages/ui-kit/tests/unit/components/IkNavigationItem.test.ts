import { IkFlex } from '@ui/components/IkFlex';
import { IkIcon } from '@ui/components/IkIcon';
import { IkNavigationItem } from '@ui/components/IkNavigation';
import { IkPopover } from '@ui/components/IkPopover';
import { getRouterMock } from '@ui/tests/mocks/router';
import type { RenderOptions } from '@ui/tests/tools/utils';
import { renderShallow } from '@ui/tests/tools/utils';

const defineRouter = () => {
    return getRouterMock({
        routes: [
            {
                path: '/',
                name: 'IkRoute1',
                component: {},
            },
            {
                path: '/route2',
                name: 'IkRoute2',
                component: {},
            },
            {
                path: '/route3/:id',
                name: 'IkRoute3',
                component: {},
            },
        ],
    });
};

describe('IkNavigationItem', () => {
    let router = defineRouter();
    const getInstance = (options: RenderOptions<typeof IkNavigationItem> = {}) => renderShallow(IkNavigationItem, {
        global: {
            provide: { navigation: { mini: false, computed_mini: false, toggle: jest.fn() } },
        },
        router,
    }, options);

    beforeEach(async () => {
        router = defineRouter();
        await router.push('/');
    });

    it('renders with default state', () => {
        const wrapper = getInstance();

        expect(wrapper.classes()).toStrictEqual([
            'ik-navigation-item',
        ]);

        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders with default and icon slots', () => {
        const wrapper = getInstance({
            slots: {
                default: 'This is unit test',
                icon: '<i>Test</i>',
            },
        });

        expect(wrapper.text()).toContain('This is unit test');
        expect(wrapper.find('i').text()).toContain('Test');

        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders with icon prop', () => {
        const wrapper = getInstance({
            props: {
                icon: 'home:regular',
            },
        });

        expect(wrapper.findComponent(IkIcon).attributes('icon')).toBe('home:regular');
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders active', () => {
        const wrapper = getInstance({
            props: {
                active: true,
            },
        });

        expect(wrapper.classes()).toContain('ik-navigation-item--active');
    });

    it('renders linked', () => {
        const wrapper = getInstance({
            props: {
                link: 'www.test.com/test',
                target: '_blank',
            },
        });

        expect(wrapper.classes()).toContain('ik-navigation-item--clickable');
        expect(wrapper.find('a').isVisible()).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders clickable', () => {
        const wrapper = getInstance({
            props: {
                onClick: jest.fn(),
            },
        });

        expect(wrapper.classes()).toContain('ik-navigation-item--clickable');
    });

    it('renders mini', () => {
        const wrapper = getInstance({
            props: { icon: 'home' },
            global: {
                provide: {
                    navigation: { mini: true },
                },
                stubs: {
                    IkPopover,
                },
            },
        });

        expect(wrapper.find('.ik-navigation-item__icon').isVisible()).toBe(true);
        expect(wrapper.find('.ik-navigation-item__content').exists()).toBe(false);
    });

    it('closes navigation when close_nav is true', async () => {
        const clickListenerMock = jest.fn();
        const toggleMock = jest.fn();

        const wrapper = getInstance({
            props: {
                close_nav: true,
                onClick: clickListenerMock,
            },
            global: {
                provide: {
                    navigation: { toggle: toggleMock },
                },
            },
        });

        // simulate click
        await wrapper.find('.ik-navigation-item__link').trigger('click');

        expect(toggleMock).toHaveBeenCalledWith(false);
        expect(clickListenerMock).toHaveBeenCalled();
    });

    it('works with route prop as route name', async () => {
        const push = jest.spyOn(router, 'push');
        const wrapper = getInstance({
            props: {
                route: 'IkRoute2',
            },
        });

        // simulate click
        await wrapper.find('.ik-navigation-item__link').trigger('click');
        expect(push).toHaveBeenCalledWith(expect.objectContaining({ name: 'IkRoute2' }));
    });

    it('works with route prop as route object', async () => {
        const push = jest.spyOn(router, 'push');
        const wrapper = getInstance({
            props: {
                route: { name: 'IkRoute3', params: { id: 567 } },
            },
        });

        // simulate click
        await wrapper.find('.ik-navigation-item__link').trigger('click');
        expect(push).toHaveBeenCalledWith(expect.objectContaining({ name: 'IkRoute3', params: { id: 567 } }));
    });

    it('renders with circle icon', () => {
        const wrapper = getInstance({
            props: {
                circle_icon: true,
            },
        });

        expect(wrapper.classes()).toContain('ik-navigation-item--circle-icon');
    });

    it('works with active route match', async () => {
        const wrapper = getInstance({
            props: {
                active_route_match: 'IkRoute2',
            },
        });

        expect(wrapper.classes()).not.toContain('ik-navigation-item--active');

        await router.replace('/route2');

        expect(wrapper.classes()).toContain('ik-navigation-item--active');
    });

    it('works with active route match regex', async () => {
        const wrapper = getInstance({
            props: {
                active_route_match: 'IkRoute*',
            },
        });

        await router.replace({ name: 'IkRoute3', params: { id: 5 } });
        expect(wrapper.classes()).toContain('ik-navigation-item--active');

        await router.replace({ name: 'IkRoute2' });
        expect(wrapper.classes()).toContain('ik-navigation-item--active');

        await router.replace({ name: 'IkRoute1' });
        expect(wrapper.classes()).toContain('ik-navigation-item--active');
    });

    it('reders with label hidden in mini variant', async () => {
        const wrapper = getInstance({
            props: {
                hide_label: true,
            },
            global: {
                provide: { navigation: { mini: true } },
            },
            slots: { default: 'Unit test' },
        });

        expect(wrapper.text()).not.toContain('Unit test');
    });

    it('renders with label visible in normal variant', async () => {
        const wrapper = getInstance({
            props: {
                hide_label: true,
            },
            provide: { navigation: { mini: false } },
            slots: { default: 'Unit test' },
        });

        expect(wrapper.text()).toContain('Unit test');
    });

    it('renders with children in mini variant', async () => {
        const wrapper = getInstance({
            props: {},
            global: {
                provide: { navigation: { mini: true } },
            },
            slots: {
                default: 'Unit test',
                children: 'This is children slot',
            },
        });

        expect(wrapper.findComponent(IkPopover).exists()).toBeTruthy();
    });

    it('renders with children in normal variant', async () => {
        const wrapper = getInstance({
            props: {},
            global: {
                provide: { navigation: { mini: false } },
            },
            slots: {
                default: 'Unit test',
                children: 'This is children slot',
            },
        });

        expect(wrapper.findComponent(IkPopover).exists()).toBeFalsy();
        expect(wrapper.text()).toContain('This is children slot');
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders with append_icon param', async () => {
        const wrapper = getInstance({
            props: {
                append_icon: 'chevron-right:light',
            },
            global: {
                stubs: { IkFlex },
            },
            slots: { default: 'Unit test' },
        });

        expect(wrapper.findComponent(IkFlex).exists()).toBeTruthy();
        expect(wrapper.findComponent(IkIcon).exists()).toBeTruthy();
        expect(wrapper.findComponent(IkIcon).attributes('icon')).toBe('chevron-right:light');
        expect(wrapper.findComponent(IkFlex).text()).toBe('Unit test');

        expect(wrapper.element).toMatchSnapshot();
    });
});
