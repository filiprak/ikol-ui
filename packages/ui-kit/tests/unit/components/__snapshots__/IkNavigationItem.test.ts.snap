// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IkNavigationItem renders linked 1`] = `
<div
  class="ik-navigation-item ik-navigation-item--clickable"
>
  <a
    class="ik-navigation-item__link"
    href="www.test.com/test"
    target="_blank"
  />
  <!---->
</div>
`;

exports[`IkNavigationItem renders with append_icon param 1`] = `
<div
  class="ik-navigation-item"
>
  <div
    class="ik-navigation-item__link"
  >
    <div
      class="ik-navigation-item__content"
    >
      <div
        class="ik-flex ik-flex--justify-space"
      >
        
        Unit test
        
        <ik-icon-stub
          circle="false"
          class="ik-ml-3"
          disabled="false"
          icon="chevron-right:light"
          tag="i"
        />
      </div>
    </div>
  </div>
  <!---->
</div>
`;

exports[`IkNavigationItem renders with children in normal variant 1`] = `
<div
  class="ik-navigation-item"
>
  <div
    class="ik-navigation-item__link"
  >
    <div
      class="ik-navigation-item__content"
    >
      
      Unit test
      
    </div>
    <ik-button-stub
      active="false"
      block="false"
      circle="true"
      class="ik-navigation-item__arrow"
      design="accent"
      disabled="false"
      fab="false"
      filled="false"
      flat="true"
      icon="chevron-down:light"
      loading="false"
      outline="false"
      round="false"
      separate="false"
      skeleton="false"
      tag="button"
      type="button"
    />
  </div>
  <div
    class="ik-navigation-item__children"
    style="display: none;"
  >
    
    This is children slot
    
  </div>
</div>
`;

exports[`IkNavigationItem renders with default and icon slots 1`] = `
<div
  class="ik-navigation-item"
>
  <div
    class="ik-navigation-item__link"
  >
    <div
      class="ik-navigation-item__icon"
    >
      
      <i>
        Test
      </i>
      
    </div>
    <div
      class="ik-navigation-item__content"
    >
      
      This is unit test
      
    </div>
  </div>
  <!---->
</div>
`;

exports[`IkNavigationItem renders with default state 1`] = `
<div
  class="ik-navigation-item"
>
  <div
    class="ik-navigation-item__link"
  />
  <!---->
</div>
`;

exports[`IkNavigationItem renders with icon prop 1`] = `
<div
  class="ik-navigation-item"
>
  <div
    class="ik-navigation-item__link"
  >
    <div
      class="ik-navigation-item__icon"
    >
      <ik-icon-stub
        circle="false"
        disabled="false"
        icon="home:regular"
        tag="i"
      />
    </div>
  </div>
  <!---->
</div>
`;
